File: day81-90/day90/1957. Delete Characters to Make Fancy String.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        stack=[]
        for char in s:
            if (len(stack)>=2 
                and stack[-1]==stack[-2] 
                and stack[-1]==char):
                continue
            stack.append(char)
        return "".join(stack)
    

sol=Solution()
print(sol.makeFancyString("aaabaaaa"))

File: day81-90/day90/1957. Delete Characters to Make Fancy String.php
<?php
class Solution {

/**
 * @param String $s
 * @return String
 */
function makeFancyString($s) {
    $length = strlen($s);
    $stack=[];
    for($i=0;$i<$length;$i++){
        $stack_length=count($stack);
        if(
            $stack_length>=2 
                &&
            $stack[$stack_length-2]==$stack[$stack_length-1] 
            &&
            $stack[$stack_length-1]==$s[$i]
            ){
            continue;
        }
        $stack[] = $s[$i];
    }
    return implode('',$stack); 
}
}

$sol=new Solution();
echo ($sol->makeFancyString("aaabaaaaaab"));